src/dpl/src/Place.cpp

```diff
@@ -946,6 +972,39 @@ PixelPt Opendp::binSearch(int x, const Cell* cell, int bin_x, int bin_y) const
     }
   } else {
     for (int i = 0; i < bin_search_width_; i++) {
+      Point p(bin_x + i, bin_y);
+      p.setX(p.getX() * site_width_);
+      p.setY(p.getY() * row_height);
+      if (cell->region_) {
+        // debugPrint(logger_,
+        //            DPL,
+        //            "detailed",
+        //            1,
+        //            "Point {} {} Cell {} has region x[{} {}] and y[{} {}]",
+        //            p.getX(),
+        //            p.getY(),
+        //            cell->name(),
+        //            cell->region_->xMin(),
+        //            cell->region_->xMax(),
+        //            cell->region_->yMin(),
+        //            cell->region_->yMax());
+        if (!cell->region_->intersects(p)) {
+          debugPrint(logger_,
+                     DPL,
+                     "detailed",
+                     2,
+                     "point and region doesn't interesect!");
```

Comment:
What is being checked?

Reply:
instead of doing the rtree query, I check if the initial point intersects with the region or not. If not, I simply skip the rest of the logic. If it does, we check if the whole cell will fit inside the region.